tests/performance/profiling.py

```diff
@@ -0,0 +1,77 @@
+import timeit
+import tracemalloc
+from multiprocessing import Process, Value
+
+import pandas as pd
+
+
+def _profile_time(method, dataset, iterations=100):
+    total_time = 0
+    for _ in range(iterations):
```

Comment:
I now realized that we are iterating here!
So yes, it seems it makes sense to profile time and memory separately, unless we would want to also average the memory consumed over the iterations.

On the other hand, if we decide to do a `deepcopy` maybe it should be happening here, before capturing the start time, rather than outside before calling this function. Also, since the `deepcopy` is needed only for `fit`, it should be made optional with an argument.